client/src/components/Leaflet.js

```diff
@@ -14,13 +16,29 @@ import "leaflet.markercluster/dist/MarkerCluster.css"
 import "leaflet.markercluster/dist/MarkerCluster.Default.css"
 import "leaflet/dist/leaflet.css"
 import { Location } from "models"
+import GeoLocation from "pages/locations/GeoLocation"
 import PropTypes from "prop-types"
 import React, { useCallback, useEffect, useRef, useState } from "react"
+import ReactDOM from "react-dom"
+import MARKER_FLAG_BLUE from "resources/leaflet/marker-flag-blue.png"
 import MARKER_ICON_2X from "resources/leaflet/marker-icon-2x.png"
 import MARKER_ICON from "resources/leaflet/marker-icon.png"
 import MARKER_SHADOW from "resources/leaflet/marker-shadow.png"
 import Settings from "settings"
 
+const allLocationsQuery = gql`
+  query {
+    locationList(query: { pageSize: 0 }) {
+      list {
+        uuid
+        name
```

Comment:
I suggest to also display the name of the locations (when not clustered) on the map